### ISSUE TITLE
Bug: no-unused-vars should not be flagged on imports used in `@link` etc.

### ISSUE THREAD

**Issue by LukeAbby** [typescript-eslint/typescript-eslint#10873]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.7.2&fileType=.tsx&code=KYDwDg9gTgLgBDAnmYcCiICGBbMAbVAXjgEYAmAZgG4BYAKHoEtdp4kU4BvdLXAuTAGc4AGUYA7ANbAAJgBVkqAL5wAZlAjY4AIgB0AeglgArjG316%2BgFRX6cK1wACeCZNGvZClCtXQ4M4BhMRjxBXTsrfSA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tidmUQAmtAG4BDaKgyRE0aB2iRwYAL4hlQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3QAacDQD00qPFgATDugBK6ErFRsCxClCJL0RdAA88kKQF8QVoA&tokens=false

### Repro Code

```TypeScript
import { LinkedType } from "./someFile"

/**
 * {@link LinkedType}
 */
```

### ESLint Config

```javascript
{
  "rules": {
    "@typescript-eslint/no-unused-vars": "error"
  }
}
```

### tsconfig

```jsonc

```

### Expected Result

I expected no error because while the import is unused outside of `@link`s I consider this a valid usage of the import. Notably tsc itself recognizes this pattern and does not give an unused variable warning. Switching all links to the much less elegant `{@link import("./someFile").LinkedType | LinkedType}` seems unideal when it's done many times and introducing a helper type just reintroduces the no-unused-vars error.

### Actual Result

The error `'LinkedType' is defined but never used.` appears.

### Additional Info

In TSDoc I believe the applicable tags are `@link`, `@inheritDoc`, and `@see`. In JSDoc as a whole it applies to any tags that could include a type. I believe that would be `@type`, `@param`, `@typedef`, `@returns`, `@property`, `@this`, `@throws`, `@yields`, and synonyms.

Important note: I strongly believe this should only apply to _imports._ This is because if you happen to write `{@link SomeUnusedThing}` in the same file `SomeUnusedThing` is defined it could mask the fact that `SomeUnusedThing` is, well, unused.

**Comment by bradzacher** [comment]
We do not support JSDoc comments.

Duplicate of https://github.com/typescript-eslint/typescript-eslint/issues/9435, #8902, #5017, #6967, #7944, #8258

Please use the search.
